server/client.go

```diff
@@ -101,7 +108,6 @@ func init() {
 func (c *client) initClient(tlsConn bool) {
```

Comment:
Wondering if Client should be an interface instead of trying to overload the actual struct. This is longer term thinking.